python/lsst/dax/imgserv/getimage/imageget.py

```diff
@@ -0,0 +1,625 @@
+                                  afwGeom.Point2D(ra2, dec2))
+            # convert from deg to arcsec
+            w = box.getWidth()*3600
+            h = box.getHeight()*3600
+            # compute the arithmetic center (ra, dec) of the range
+            ra = (ra1 + ra2) / 2
+            dec = (dec1 + dec2) / 2
+            cutout = self.cutout_from_nearest(ra, dec, w, h, "arcsec", filt)
+            return cutout
+        elif shape == "POLYGON":
+            vertices = []
+            pos_items.pop(0)
+            for long, lat in zip(pos_items[::2], pos_items[1::2]):
+                pt = afwGeom.Point2D(float(long), float(lat))
+                vertices.append(pt)
+            polygon = afwGeom.Polygon(vertices)
+            center = polygon.calculateCenter()
+            ra, dec = center.getX(), center.getY()
+            # afw limitation: can only return the bbox of the polygon
+            bbox = polygon.getBBox()
+            # convert from 'deg' to 'arcsec'
+            w = bbox.getWidth()*3600
+            h = bbox.getHeight()*3600
+            cutout = self.cutout_from_nearest(ra, dec, w, h, "arcsec", filt)
+            return cutout
+
+    @staticmethod
+    def cutout_from_exposure(src_image, ra, dec, width, height, unit="pixel"):
+        """ Get the Exposure cutout including wcs headers.
+
+        Parameters
+        ----------
+        src_image : `afwImage.Exposure`
+            the source image.
+        ra : `float`
+            in degrees.
+        dec : `float`
+            in degrees.
+        width : int
+            in pixels.
+        height : int
+            in pixels
+
+        Returns
+        -------
+        cutout : `afwImage.Exposure`
+
+        """
+        if unit == "arcsec":
+            # pixel scale is defined as Angle/pixel
+            wcs = src_image.getWcs()
+            ps = wcs.getPixelScale().asArcseconds()
+            if ps != 0:
+                width = width / ps
+                height = height / ps
+            else:
+                return "Unexpected: pixel scale = 0"
+        center = afwGeom.SpherePoint(ra, dec, afwGeom.degrees)
+        size = afwGeom.Extent2I(width, height)
+        cutout = src_image.getCutout(center, size)
+        return cutout
+
+    def _cutout_by_data_id(self, data_id, ra, dec, width, height, unit="pixel"):
+        # check to see if q_results is empty
+        if data_id is None:
+            return None
+        # Return an image by data ID through the butler.
+        image = self._image_from_butler(data_id)
+        if image is None:
+            # @todo html error handling see DM-1980
+            return None
+        if isinstance(image, afwImage.Exposure):
+            cutout = ImageGetter.cutout_from_exposure(image, ra, dec, width,
+                                                      height, unit)
+        else:
+            cutout = self._apply_cutout(data_id, image, ra, dec, width, height,
```

Comment:
So if image is an afwImage.Exposure, call cutout_from_exposure.  If it's not, then call _apply_cutout, but _apply_cutout by its docs also takes src_img to be an afwImage.Exposure?